corehq/apps/userreports/views.py

```diff
@@ -1344,12 +1346,36 @@ def rebuild_data_source(request, domain, config_id):
         )
     )
 
-    rebuild_indicators.delay(config_id, request.user.username, domain=domain)
+    rebuild_indicators.delay(config_id, request.user.username, domain=domain, source='edit_data_source_rebuild')
+    _report_ucr_rebuild_metrics(domain, config, 'rebuild_datasource')
     return HttpResponseRedirect(reverse(
         EditDataSourceView.urlname, args=[domain, config._id]
     ))
 
 
+def _report_ucr_rebuild_metrics(domain, config, action):
+    metrics_counter(f'commcare.ucr.{action}.count', tags={'domain': domain})
+    metrics_gauge(
+        f'commcare.ucr.{action}.columns.count',
+        len(config.get_columns()),
+        tags={'domain': domain}
+    )
```

Comment:
>  think unless you tag the data source in some way, the value for the domain will change with every data source you send a metric for. I

That is right and a valid concern.

The original requirement was for including `datasource_id`.  It was my recommendation to not include it considering below factors:
- The aim is to keep the cardinality(variance) to minimum in general to avoid increase in Datadog costs. Note that this is not a restriction though. 
- Based on my the requirements, the idea is see the general trend of `how many a columns a datasource has while being rebuilt` instead of a report with datasource and their columns count. Since the ticket is specificaly focused to track the usage of **Rebuild** and **Rebuild in Place** functionality, a general trend (e.g `average`) seems to be a reasonable insight.
- For the DataDog visualisation, the intention is to show a average of above metric with an option to segment/filter by domain.

Let me know if this is something that sounds reasonable.

Reply:
Thanks. That does sound reasonable. Just taking the average is an excellent approach.